### ISSUE TITLE
C:\Strawberry\c\bin\cmake.exe could not find libexpat-1.dll

### ISSUE THREAD

**Issue by code4lala** [StrawberryPerl/Perl-Dist-Strawberry#219]
![image](https://github.com/user-attachments/assets/220a7d0c-7c3b-4d71-8be5-d44c67d77b27)

if I copy `C:\Strawberry\c\bin\libexpat-1__.dll` to `C:\Strawberry\c\bin\libexpat-1.dll`, copy the dll file and remove two underlines, cmake.exe would work.

**Comment by shawnlaffan** [comment]
I am not sure this is a Strawberry Perl issue.  

cmake is not distributed with Strawberry Perl 5.40 due to clashes with other packaging systems.  

If you are deriving it from elsewhere then you need to ensure it has access to the relevant DLLs it depends on.

**Comment by code4lala** [comment]
The cmake is from the strawberry perl package, I downloaded the strawberry perl msi from the official site:

https://strawberryperl.com/

https://github.com/StrawberryPerl/Perl-Dist-Strawberry/releases/download/SP_54001_64bit_UCRT/strawberry-perl-5.40.0.1-64bit.msi

**Comment by mohawk2** [comment]
I think cmake being included is a great idea. It just sounds like it needs this double-underscore business including in Strawberry's cmake-visible packaging info.

**Comment by shawnlaffan** [comment]
I stand corrected.  It comes from winlibs and should have been removed before we cut the release.  

We could look at building our own cmake but there is an Alien::cmake that works well.  That also avoids issues such as SP 5.36 which had an old version of cmake that got in the way of another toolchain (I don't remember all the details at the moment).

See also #154

**Comment by mohawk2** [comment]
Auditing things sounds like a good idea! But this doesn't seem like a problem with building cmake, only with the info it finds within the distro.

@code4lala can you take us through the steps you took to find this problem? Did you make a local cmake distro, and if so, can you share all the parts of it so we could try that out?

**Comment by code4lala** [comment]
@mohawk2 sure, it's easy! just download https://github.com/StrawberryPerl/Perl-Dist-Strawberry/releases/download/SP_54001_64bit_UCRT/strawberry-perl-5.40.0.1-64bit.msi and install it! then go to explorer and navigvate to C:\Strawberry\c\bin\, double click cmake.exe

**Comment by mohawk2** [comment]
@code4lala I don't think that is supposed to do very much. It's not a GUI application. When you say it "worked" after the DLL copying, what do you mean?

**Comment by code4lala** [comment]
@mohawk2 after copying, cmake would not pop up an error window, and it works, I can use it to build my project.

**Comment by mohawk2** [comment]
@code4lala Can you try undoing the copy/rename, and then using it to build your project? I want to test whether SP5.40 can use its supplied cmake to build projects, and double-clicking it in Explorer isn't part of that.

**Comment by code4lala** [comment]
@mohawk2 undoing the copy/rename, and then using it to build your project -> cmake just exit without printing anything.
@shawnlaffan the msi installation didn't report any error.

**Comment by shawnlaffan** [comment]
There have been no updates to this issue for a while now.  

My suspicion is that it is an issue with the user system, possibly mixed paths.  Such a situation will not provide an error on installation.  

I'll close this in a few days or when I next do a sweep of issues.

**Comment by shawnlaffan** [comment]
Closing as noted.
